Editor/HPUIInteractorConeRayAnglesEditor.cs

```diff
@@ -0,0 +1,163 @@
+using UnityEngine;
+using UnityEditor;
+using ubco.ovilab.HPUI.Interaction;
+using UnityEngine.XR.Hands;
+
+namespace ubco.ovilab.HPUI.Editor
+{
+    [CanEditMultipleObjects]
+    [CustomEditor(typeof(HPUIInteractorConeRayAngles), true)]
+    public class HPUIInteractorConeAnglesEditor: UnityEditor.Editor
+    {
+        private HPUIInteractorConeRayAngles t;
+
+        private float offsetValue;
+        private bool applyToAll;
+        private bool overwriteThisAsset;
+        private XRHandJointID selectedPhalange = XRHandJointID.IndexDistal;
+
+        // storing all baseline thresholds
+        private Dictionary<XRHandJointID, List<float>> baselineThresholds = new Dictionary<XRHandJointID, List<float>>();
+
+        private const string DONT_ASK_FOR_OVERRIDE_EDITOR_PREF_KEY = "ubco.ovilab.HPUI.Interaction.ConeThresholdsEditor.DontAskForOverride";
+        private bool userSelection;
+        private string newAssetFileSaveName;
+
+        private bool advancedSettingsFoldout;
+
+        private const string OFFSET_TOOLTIP = "The uniform offset to apply to the ray thresholds for a given phalange or all phalanges. Note that it always applies it from the base values, which are derived upon the first time the asset is selected in this Unity session. So setting it to 2 and then 1 won't increment it by 3, the second operation will only increment it by 1 from the base amount. Consequently, setting it to 0 will reset it to whatever the original values were.";
+
+        protected void OnEnable()
+        {
+            t = target as HPUIInteractorConeRayAngles;
+            if (!t) return;
+            foreach ((XRHandJointID phalange, List<HPUIInteractorRayAngle> rays) in t.ActiveFingerAngles)
+            {
+                if (!baselineThresholds.ContainsKey(phalange) && rays.Count > 0)
+                {
+                    baselineThresholds[phalange] = rays.Select(ray => ray.RaySelectionThreshold).ToList();
+                }
+            }
+        }
+
+        public override void OnInspectorGUI()
+        {
+            base.OnInspectorGUI();
+
+            EditorGUILayout.LabelField("", GUI.skin.horizontalSlider);
+
+            advancedSettingsFoldout = EditorGUILayout.Foldout(advancedSettingsFoldout, "Advanced Operations", toggleOnLabelClick:true);
+            if (advancedSettingsFoldout)
+            {
+                EditorGUILayout.LabelField("Apply Ray Selection Threshold Offset", EditorStyles.boldLabel);
+                offsetValue = EditorGUILayout.FloatField(new GUIContent("Offset", OFFSET_TOOLTIP), offsetValue);
+                applyToAll = EditorGUILayout.Toggle(new GUIContent("Apply to All", "apply offset to all phalanges at once"), applyToAll);
+                overwriteThisAsset = EditorGUILayout.Toggle("Overwrite This Asset", overwriteThisAsset);
+
+                EditorGUI.BeginDisabledGroup(applyToAll);
+                selectedPhalange = (XRHandJointID)EditorGUILayout.EnumPopup("Phalange", selectedPhalange);
```

Comment:
Given that only a subset of the `XRHandJointID` enum is used, consider using generic Popup and parse it to the XRHandJointID